ebur128/ebur128.c

```diff
@@ -675,6 +675,74 @@ EBUR128_FILTER(int, INT_MIN, INT_MAX)
 EBUR128_FILTER(float, -1.0f, 1.0f)
 EBUR128_FILTER(double, -1.0, 1.0)
 
+#define EBUR128_FILTER_PLANAR(type, min_scale, max_scale)                      \
```

Comment:
Instead of duplicating all this code, you could do access to the `src` samples via a macro that is defined differently for interleaved/planar. Something like

```cpp
#define GET_SAMPLE_INTERLEAVED(src, channels, c, s) ((src)[(channels) * (s) + (c)])
#define GET_SAMPLE_PLANAR(src, channels, c, s) ((src)[(c)][(s)])
```

Make such macro an argument to the existing `FILTER` / `ADD_FRAMES` macros and make use of them in there.